solr/test-framework/src/java/org/apache/solr/util/TestHarness.java

```diff
@@ -186,6 +187,9 @@ public TestHarness(NodeConfig config, CoresLocator coresLocator) {
   }
 
   public static NodeConfig buildTestNodeConfig(Path solrHome) {
+    if ("true".equals(System.getProperty("solr.tests.loadSolrXml"))) {
+      return SolrXmlConfig.fromSolrHome(solrHome, new Properties());
+    }
```

Comment:
Just wanted to call this out -- am I missing something, is there a better way to do this? I'd be surprised if this were the first time anyone wanted to actually build and query a core in a way that calls for custom functionality in `solr.xml` ... but I haven't been able to find any precedent.

There's lots of precedent for directly building a NodeConfig; but it looks like the way TestHarness builds its NodeConfig (like, to support actual indexing and querying) ignores `solr.xml` here in `buildTestNodeConfig()`.

Aside from this last minor question, this should now be ready to commit.

Reply:
FYI I'm trying to get us off of using TestHarness; it's very legacy.  I recommend avoiding it for new tests.
https://issues.apache.org/jira/browse/SOLR-11872

Reply:
Sorry about that! I'm curious, if it's not too much trouble: is there a test you'd recommend as demonstrating a more modern pattern -- particularly one that modifies `solr.xml` _and_ goes all the way through indexing/querying? It was hard enough for me to track down any way to do this at all, let alone the right way.

I don't mind circling back to improve this, since it's in the front of my mind now.

Reply:
See #1886